internal/bzlmod/go_deps.bzl

```diff
@@ -42,10 +64,28 @@ def _go_deps_impl(module_ctx):
 
     outdated_direct_dep_printer = print
     for module in module_ctx.modules:
-        # Parse the go_dep.config tag of the root module only.
+        def fail_unless_overriding_tolerated(attribute_name, tag_class):
+            if not (module.is_root or _tolerate_overrides_in_module(module)):
+                fail("""\
+Overriding the "{attribute_name}" attribute in a "go_deps.{tag_class}" tag
+is forbidden in non-root Bazel modules, but module "{module_name}" requests it.
+
+If you need this override for a Bazel module that will be available in the
+Bazel Central Registry, with other modules depending on it, please file an issue
```

Comment:
The "with other modules depending on it" is a bit redundant as that's really the only point of adding something to the registry.

We may also want to say "public registry (such as the Bazel Central Registry)" instead as there could very well be others in the future.